addons/web/static/src/js/widgets/date_picker.js

```diff
@@ -79,8 +79,11 @@ var DateWidget = Widget.extend({
             var oldValue = this.getValue();
             this._setValueFromUi();
             var newValue = this.getValue();
-
-            if (!oldValue !== !newValue || oldValue && newValue && !oldValue.isSame(newValue)) {
+            if(oldValue && newValue){
```

Comment:
i would prefer something like this:
`        var isSame = !oldValue !== !newValue;
        if (oldValue && newValue) {
            var formattedOldValue = oldValue.format(time.getLangDatetimeFormat());
            var formattedNewValue = newValue.format(time.getLangDatetimeFormat())
            if (formattedNewValue !== formattedOldValue) {
                isSame = false;
            }
        }
        if (isSame) {
            ...
        }
`